src/compile/axis/encode.ts

```diff
@@ -27,8 +27,17 @@ export function labels(model: UnitModel, channel: SpatialScaleChannel, specified
     labelsSpec.text =  {
       signal: timeFormatExpression('datum.value', fieldDef.timeUnit, axis.format, config.axis.shortTimeLabels, config.timeFormat, isUTCScale)
     };
+  } else if ((fieldDef.type === NOMINAL || fieldDef.type === ORDINAL) && axis.format) {
+    if (fieldDef['formatType'] === 'number') {
```

Comment:
Why `fieldDef['formatType']` ?   First of all, trying to access value with string is a hack, which can be risky.  In this case, you add `formatType` to axis but access it from fieldDef, so this is definitely wrong.  Maybe you should have access this from `axis.formatType` instead?